### ISSUE TITLE
Basic auth for HTTP Listener broken 

### ISSUE THREAD

**Issue by pcunning** [hubotio/hubot#1469]
When trying to use basic auth with the Express HTTP Listener (by setting EXPRESS_USER/EXPRESS_PASSWORD) Hubot crashes with the following error because basicAuth is not available in Express 4.x.

```
TypeError: express.basicAuth is not a function
    at Robot.setupExpress (./node_modules/hubot/src/robot.js:440:23)
```
https://github.com/hubotio/hubot/blob/85284750c878205aeae7338513961b29f0a1d18f/src/robot.js#L440

The version of Express Hubot uses was upgraded in #1453 and released in 3.1.0 breaking this feature.

**Comment by MingStar** [comment]
I encountered the same problem today, could the fix #1470  be merged?

**Comment by MingStar** [comment]
Or is basic auth not recommended?
